cvxpy/lin_ops/canon_backend.py

```diff
@@ -1103,6 +1090,318 @@ def _to_dense(x):
+        assert is_param_free_lhs
+        # dtype is important here, will do integer division if data is of dtype "int" otherwise.
+        lhs.data = np.reciprocal(lhs.data, dtype=float)
+
+        def div_func(x, _p):
+            return lhs.multiply(x)
+
+        return view.accumulate_over_variables(div_func, is_param_free_function=is_param_free_lhs)
+
+    @staticmethod
+    def diag_vec(lin: LinOp, view: StackedSlicesTensorView) -> StackedSlicesTensorView:
+        assert lin.shape[0] == lin.shape[1]
+        k = lin.data
+        rows = lin.shape[0]
+        total_rows = int(lin.shape[0] ** 2)
+
+        def func(x, _p):
+            shape = list(x.shape)
+            shape[0] = total_rows
+            x = x.tocoo()
+            if k == 0:
+                new_rows = (x.row * (rows + 1)).astype(int)
+            elif k > 0:
+                new_rows = (x.row * (rows + 1) + rows * k).astype(int)
+            else:
+                new_rows = (x.row * (rows + 1) - k).astype(int)
+            return sp.csr_matrix((x.data, (new_rows, x.col)), shape)
+
+        view.apply_all(func)
+        return view
+
+    @staticmethod
+    def get_stack_func(total_rows: int, offset: int) -> Callable:
+        def stack_func(tensor, _p):
+            coo_repr = tensor.tocoo()
+            new_rows = (coo_repr.row + offset).astype(int)
+            return sp.csr_matrix((coo_repr.data, (new_rows, coo_repr.col)),
+                                 shape=(int(total_rows), tensor.shape[1]))
+
+        return stack_func
+
+    def rmul(self, lin: LinOp, view: StackedSlicesTensorView) -> StackedSlicesTensorView:
+        """
+        Multiply view with constant data from the right.
+        """
+        # Note that even though this is rmul, we still use "lhs", as is implemented via a
+        # multiplication from the left in this function.
+        lhs, is_param_free_lhs = self.get_constant_data(lin.data, view, column=False)
+
+        arg_cols = lin.args[0].shape[0] if len(lin.args[0].shape) == 1 else lin.args[0].shape[1]
+
+        if is_param_free_lhs:
+            if len(lin.data.shape) == 1 and arg_cols != lhs.shape[0]:
+                lhs = lhs.T
+            reps = view.rows // lhs.shape[0]
+            stacked_lhs = sp.kron(lhs.T, sp.eye(reps, format="csr"))
+
+            def func(x, _p):
+                return stacked_lhs @ x
+        else:
+            p = view.rows // next(iter(lhs.values())).shape[-1]
+            lhs_shape = next(iter(lhs.values())).shape
+            lhs_shape = (lhs_shape[0] // p, lhs_shape[1])
+            if len(lin.data.shape) == 1 and arg_cols != lhs_shape[0]:
+                # Example: (n,n) @ (n,), we need to interpret the rhs as a column vector,
+                # but it is a row vector by default, so we need to transpose
+                lhs_shape = next(iter(lhs.values())).shape
+                lhs_shape = (lhs_shape[0], lhs_shape[1] // p)
+                for param_id, param_mat in lhs.items():
```

Comment:
there is probably a better way to write this, but essentially I am just doing similar sub-indexing to mul (but for the new shape). Then I simply transpose and stack the slices vertically, it is sort of semantically equivalent to [v_I.T for v_I in v]